iothub/device/src/IotHubModuleClient.cs

```diff
@@ -233,70 +221,27 @@ public Task<DirectMethodResponse> InvokeMethodAsync(string deviceId, DirectMetho
         /// <param name="methodRequest">The details of the method to invoke.</param>
         /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
         /// <returns>The result of the method invocation.</returns>
-        /// <exception cref="InvalidOperationException">Thrown if IotHubModuleClient instance is not opened already.</exception>
+        /// <exception cref="InvalidOperationException">Thrown if the client instance is not opened already.</exception>
         /// <exception cref="OperationCanceledException">Thrown when the operation has been canceled.</exception>
         /// <exception cref="ObjectDisposedException">When the client has been disposed.</exception>
         public Task<DirectMethodResponse> InvokeMethodAsync(string deviceId, string moduleId, DirectMethodRequest methodRequest, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNull(methodRequest, nameof(methodRequest));
-            if (_isDisposed)
-            {
-                throw new ObjectDisposedException(nameof(IotHubModuleClient));
-            }
+            cancellationToken.ThrowIfCancellationRequested();
 
             return InvokeMethodAsync(GetModuleMethodUri(deviceId, moduleId), methodRequest, cancellationToken);
         }
 
-        /// <inheritdoc/>
-        protected override void Dispose(bool disposing)
-        {
-            if (disposing)
-            {
-                _moduleReceiveMessageSemaphore?.Dispose();
-            }
-
-            // Call the base class implementation.
-            base.Dispose(disposing);
-        }
-
         private async Task<DirectMethodResponse> InvokeMethodAsync(Uri uri, DirectMethodRequest methodRequest, CancellationToken cancellationToken = default)
         {
-            HttpClientHandler httpClientHandler = null;
-            Func<object, X509Certificate, X509Chain, SslPolicyErrors, bool> customCertificateValidation = _certValidator.GetCustomCertificateValidation();
-
-            try
-            {
-                var pipelineContext = new PipelineContext
-                {
-                    IotHubConnectionCredentials = IotHubConnectionCredentials,
-                    HttpOperationTransportSettings = new IotHubClientHttpSettings(),
-                };
+            methodRequest.PayloadConvention = _clientOptions.PayloadConvention;
+            DirectMethodResponse result = await InnerHandler.InvokeMethodAsync(methodRequest, uri, cancellationToken).ConfigureAwait(false);
 
-                if (customCertificateValidation != null)
-                {
-                    httpClientHandler = new HttpClientHandler
-                    {
-                        ServerCertificateCustomValidationCallback = customCertificateValidation,
-                        SslProtocols = pipelineContext.HttpOperationTransportSettings.SslProtocols,
```

Comment:
The certificate validator and ssl protocols are set on the Http client handler directly